### PR TITLE
clean up http client

### PR DESCRIPTION
**Pull request by wangxinyi7**
### Description

Though there are lots of differences in the PR, what I did was just removing the HTTP client related things.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: command/resource/apply/apply_test.go
Comment (loshz) [inline command/resource/apply/apply_test.go:130]: Do we need to do this?

File: command/resource/apply/apply_test.go
Comment (loshz) [inline command/resource/apply/apply_test.go:181]: Do we need to do this?

File: command/resource/apply/apply_test.go
Comment (wangxinyi7) [reply command/resource/apply/apply_test.go]: good catch! added the check

File: command/resource/apply/apply_test.go
Comment (wangxinyi7) [reply command/resource/apply/apply_test.go]: good catch! added the check

**Review by loshz — APPROVED** [review]
The changes LGTM, but I do have little context. It's worth getting @analogue to give their stamp of approval, too.

**Comment by wangxinyi7** [comment]
> The changes LGTM, but I do have little context. It's worth getting @analogue to give their stamp of approval, too.

Yup, sounds good, I'll wait for his input as well, thanks for your review!

File: command/resource/apply/apply.go
Comment (analogue) [inline command/resource/apply/apply.go:139]: You'll probably have to remove all references to `PeerName` before/during merge since it just got removed from the tenancy struct across the entire codebase.

**Review by analogue — APPROVED** [review]

File: command/resource/apply/apply.go
Comment (wangxinyi7) [reply command/resource/apply/apply.go]: Thanks for letting me know! Double checked to make sure I don't have legacy `Peer` stuff in the resource cli command area

**Pull request merged**
